sfputil/main.py

```diff
@@ -864,6 +880,100 @@ def on(port_name):
     """Enable low-power mode for SFP transceiver"""
     set_lpmode(port_name, True)
 
+# 'show loopback' subcommand
+@show.command()
+@click.option('-p', '--port', metavar='<port_name>', help="Display SFP low-power mode status for port <port_name> only")
+def loopback(port):
+    """Display loopback mode of SFP transceiver(s)"""
+    # Create a list containing the logical port names of all ports we're interested in
+    if port is None:
+        logical_port_list = platform_sfputil.logical
+    else:
+        if platform_sfputil.is_logical_port(port) == 0:
+            click.echo("Error: invalid port '{}'\n".format(port))
+            print_all_valid_port_values()
+            sys.exit(ERROR_INVALID_PORT)
+
+        logical_port_list = [port]
+
+    click.echo("{:<20}{}".format("Port", "Loopback Modes"))
+    click.echo("------------------  ------------------------")
+    for logical_port in logical_port_list:
+        physical_port_list = logical_port_name_to_physical_port_list(logical_port)
+        if physical_port_list is None:
+            click.echo("Error: No physical ports found for logical port '{}'".format(logical_port))
+            return
+        physical_port = physical_port_list[0]
+        modes = []
+        try:
+            if platform_chassis.get_sfp(physical_port).get_cmis_loopback('mo'):
+                modes.append('mo')
+            if platform_chassis.get_sfp(physical_port).get_cmis_loopback('mi'):
+                modes.append('mi')
+            if platform_chassis.get_sfp(physical_port).get_cmis_loopback('ho'):
+                modes.append('ho')
+            if platform_chassis.get_sfp(physical_port).get_cmis_loopback('hi'):
+                modes.append('hi')
+        except NotImplementedError:
+            click.echo("This functionality is currently not implemented for this platform")
+            sys.exit(ERROR_NOT_IMPLEMENTED)
+        click.echo("{:<20}{}".format(logical_port, ",".join(modes)))
+
+# 'loopback' subgroup
+@cli.group()
+def loopback():
+    """Setting loopback mode for SFP transceiver"""
+    pass
+
+# Helper method for setting loopback mode
+def set_loopback(port, mode, enable):
+    lp_modes = [
+        'mi', 'media_input',
+        'mo', 'media_output',
+        'hi', 'host_input',
+        'ho', 'host_output'
+    ]
+
+    # Create a list containing the logical port names of all ports we're interested in
+    if port is None:
+        logical_port_list = platform_sfputil.logical
+    else:
+        if platform_sfputil.is_logical_port(port) == 0:
+            click.echo("Error: invalid port '{}'\n".format(port))
+            print_all_valid_port_values()
+            sys.exit(ERROR_INVALID_PORT)
+
+        logical_port_list = [port]
+
+    for logical_port in logical_port_list:
+        physical_port_list = logical_port_name_to_physical_port_list(logical_port)
```

Comment:
in which case there will be multiple physical ports underneath one logical port?